gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HookHandler.java

```diff
@@ -569,6 +577,22 @@ public boolean handle(final RoutingContext ctx) {
             }
         }
 
+        // 1. Check if the request method is GET
+        if (request.method() == HttpMethod.GET) {
+            String uri = request.uri();
+            String queryParam = request.getParam("q");
+            // 2. Check if the URI is for listeners or routes and has a query parameter
+            if (queryParam != null && !queryParam.isEmpty()) {
+                if (uri.contains(LISTENERS_KEY)) {
```

Comment:
Don't look for `listeners` only. This will also match other requests not related to hooks. E.g.

```
/nemo/some/url/listeners/foobar
```
The same applies for `routes`. You always have a configurable prefix like `playground/server/hooks/v1/registrations` which should be used for this.

Reply:
Validation has been implemented as suggested.